zrml/prediction-markets/src/migrations.rs

```diff
@@ -698,6 +704,255 @@ mod tests_authorized {
     }
 }
 
+use frame_support::dispatch::EncodeLike;
+use zeitgeist_primitives::types::{MarketDispute, OldMarketDispute};
+
+const PREDICTION_MARKETS_REQUIRED_STORAGE_VERSION: u16 = 7;
+const PREDICTION_MARKETS_NEXT_STORAGE_VERSION: u16 = 8;
+
+#[cfg(feature = "try-runtime")]
+type OldDisputesOf<T> = frame_support::BoundedVec<
+    OldMarketDispute<
+        <T as frame_system::Config>::AccountId,
+        <T as frame_system::Config>::BlockNumber,
+    >,
+    <T as crate::Config>::MaxDisputes,
+>;
+
+pub struct MoveDataToSimpleDisputes<T>(PhantomData<T>);
+
+impl<T: Config + zrml_simple_disputes::Config + zrml_market_commons::Config> OnRuntimeUpgrade
+    for MoveDataToSimpleDisputes<T>
+where
+    <T as zrml_market_commons::Config>::MarketId: EncodeLike<
+        <<T as zrml_simple_disputes::Config>::MarketCommons as MarketCommonsPalletApi>::MarketId,
+    >,
+{
+    fn on_runtime_upgrade() -> Weight {
+        use orml_traits::NamedMultiReservableCurrency;
+        use zeitgeist_primitives::types::Asset;
+
+        let mut total_weight = T::DbWeight::get().reads(1);
+        let pm_version = StorageVersion::get::<crate::Pallet<T>>();
+        if pm_version != PREDICTION_MARKETS_REQUIRED_STORAGE_VERSION {
+            log::info!(
+                "MoveDataToSimpleDisputes: market-commons version is {:?}, but {:?} is required",
+                pm_version,
+                PREDICTION_MARKETS_REQUIRED_STORAGE_VERSION,
+            );
+            return total_weight;
+        }
+        log::info!("MoveDataToSimpleDisputes: Starting...");
+
+        total_weight = total_weight.saturating_add(T::DbWeight::get().reads(1));
+
+        for (market_id, old_disputes) in crate::Disputes::<T>::drain() {
+            total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+
+            total_weight = total_weight.saturating_add(T::DbWeight::get().reads(1));
+            if let Ok(mut market) = <zrml_market_commons::Pallet<T>>::market(&market_id) {
+                match market.dispute_mechanism {
+                    MarketDisputeMechanism::Authorized => {
+                        // TODO Move this part of the migration (market mutation)
+                        // TODO into the OutsiderBond migration (needs to run before this migration)
+                        // TODO because it needs to iterate over all Disputes of the pm pallet
+                        // TODO in this current migration we drain all Disputes of the pm pallet
```

Comment:
Regard TODOs here and wait for OutsiderBond PR to get merged. 

Additionally add migration tests for the simple dispute Disputes transformation.

Reply:
https://github.com/zeitgeistpm/zeitgeist/pull/938/commits/06314bbc92909241477267a4e0f698412e658979